languages/elixir/exercises/concept/tuples/.docs/introduction.md

```diff
@@ -0,0 +1,67 @@
+In Elixir, a tuple is a data structure which organizes data in contiguous memory. It behaves as a static data structure, holding a fixed number of items, but without explicit names for each element. Tuples are often used in Elixir for memory read-intensive operations, since read-access of an element is a constant-time operation. They are not usually used when elements may need to be added/removed dynamically because rather than modifying the existing tuple, a new tuple is created which requires memory to be allocated upfront.
+
+In practice, tuples are created in Elixir using curly braces, and element can be individually accessed using the `elem/1` function using 0-based indexing:
+
+```elixir
+empty_tuple = {}
+one_element_tuple = {1}
+multiple_element_tuple = {1, :a, "hello"}
+
+elem(multiple_element_tuple, 2)
+# => "hello"
+```
+
+## Tuples as grouped information
+
+Tuples are often used in practice to represent grouped information.
+
+```elixir
+Float.ratio(0.25)
+# => {1, 4} indicating the numerator and denominator of the fraction ¼
+```
+
+## Pattern matching basics with tuples
+
+The use of pattern matching is dominant in assertive, idiomatic Elixir code. You might recall that `=/2` is described as a match operator rather than as an assignment operator. When elixir invokes the `=/2` function, if the pattern on the left matches the right, any variables on the left are bound, and the function returns the value of the right side. A `MatchError` is raised if there is no match.
+
+```elixir
+2 = 2
+# => 2
+# Literals can be matched if they are the same
+
+2 = 3
+# => ** (MatchError) no match of right hand side value: 3
+
+{_, denominator} = Float.ratio(0.25)
+# => {1, 4}
+# the variable `denominator` is bound to the value 4
+```
+
+> Remember, matches occur from the right side to the left side.
+
+In the last example if we don't need a variable in a pattern match, we can discard it by referencing `_`. Any variable starting with an `_` is not tracked by the runtime.
+
+## Pattern matching in named functions
+
+Pattern matching is also a useful tool when creating multiple function clauses. Pattern matching can be used on the functions' parameters which then determines which function clause to invoke -- starting from the top of the file down until the first match. Variables may be bound to patterns.
+
+```elixir
+defmodule Example do
+  def named_function(:a = variable_a) do
+    {variable_a, 1}
+  end
+
+  def named_function(:b = variable_b) do
+    {variable_b, 2}
+  end
```

Comment:
If I look at the example code, the pattern matching on the parameter is not used. Maybe this is something that could only be explained in the `after.md` document, whereas you could then simplify this example to:

```suggestion
  def named_function(:a) do
    {true, 1}
  end

  def named_function(:b) do
    {false, 2}
  end
```

Does that make sense?

Reply:
https://github.com/exercism/v3/blob/16e039d95d4bb8e57886daf9e733575bac44493e/languages/elixir/exercises/concept/tuples/.meta/example.ex#L25

Reply:
I'm not sure what you mean, I think I use the feature you describe on the line linked above.

Reply:
Ah right! I missed that. Maybe then have one function do the regular, no-assigned version and one with the assigned version?

```suggestion
  def named_function(:a) do
    {:a, 1}
  end

  def named_function(:b = variable_b) do
    {variable_b, 2}
  end
```

Reply:
I like this too, showcases that you can easily implement different behaviours!